gstools/krige/tools.py

```diff
@@ -63,16 +62,16 @@ def eval_func(func, pos, mesh_type="structured"):
     :class:`numpy.ndarray`
         Function values at the given points.
     """
-    x, y, z, dim = pos2xyz(pos, calc_dim=True)
-    if mesh_type == "structured":
-        x, y, z, axis_lens = reshape_axis_from_struct_to_unstruct(dim, x, y, z)
-    res = func(*[x, y, z][:dim])
-    if mesh_type == "structured":
-        res = reshape_field_from_unstruct_to_struct(dim, res, axis_lens)
-    return res
+    if mesh_type != "unstructured":
+        pos, shape = format_struct_pos_dim(pos, dim)
+        pos = gen_mesh(pos)
+    else:
+        pos = np.array(pos, dtype=np.double).reshape(dim, -1)
+        shape = np.shape(pos[0])
+    return np.reshape(func(*pos), shape)
 
 
-def set_condition(cond_pos, cond_val, max_dim=3):
+def set_condition(cond_pos, cond_val, dim):
```

Comment:
Would it make sense to put the argument `dim` to the first position to make it consistent with e.g. `CovModel`?

Reply:
I can do so, but this is a low-level function users shouldn't use. But following [Hyrum's Law](https://github.com/dwmkerr/hacker-laws#hyrums-law-the-law-of-implicit-interfaces) it could be a good idea to be consistent.

Reply:
Thinking about it twice, I think we should keep it this way. It is low-level and the first parameters should hold the conditions, as the function name states.